src/app/src/main/kotlin/org/vaccineimpact/api/app/Helpers.kt

```diff
@@ -23,12 +23,18 @@ fun addTrailingSlashes(req: Request, res: Response)
     }
 }
 
-fun addDefaultResponseHeaders(res: Response, contentType: String = "${ContentTypes.json}; charset=utf-8")
-    = addDefaultResponseHeaders(res.raw(), contentType = contentType)
-fun addDefaultResponseHeaders(res: HttpServletResponse, contentType: String = "${ContentTypes.json}; charset=utf-8")
+fun addDefaultResponseHeaders(req: Request,
+                              res: Response,
+                              contentType: String = "${ContentTypes.json}; charset=utf-8")
+    = addDefaultResponseHeaders(req, res.raw(), contentType)
+
+
+fun addDefaultResponseHeaders(req: Request, res: HttpServletResponse,
+                              contentType: String = "${ContentTypes.json}; charset=utf-8")
 {
     res.contentType = contentType
-    if (res.getHeader("Content-Encoding") != "gzip")
+    val gzip = req.headers("Accept-Encoding")?.contains("gzip")
+    if (gzip != null && gzip && res.getHeader("Content-Encoding") != "gzip")
```

Comment:
Can you do `gzip == true` instead of `gzip != null && gzip`?

Reply:
Oh good shout, I wanted a `.isNullOrFalse` but didn't think of that simpler option! Doh